phase/extender_test.go

```diff
@@ -196,623 +196,327 @@ func testExtender(t *testing.T, when spec.G, it spec.S) {
-						},
-						logger,
-					).DoAndReturn(
-						func(dockerfile extend.Dockerfile, _ v1.Image, _ extend.Options, _ llog.Logger) (v1.Image, error) {
-							h.AssertEq(t, dockerfile.Path, expectedDockerfileA.Path)
-							h.AssertEq(t, len(dockerfile.Args), 4)
-							h.AssertEq(t, dockerfile.Args[0].Name, "build_id")
-							_, err := uuid.Parse(dockerfile.Args[0].Value)
-							h.AssertNil(t, err)
-							h.AssertEq(t, dockerfile.Args[1].Name, "user_id")
-							h.AssertEq(t, dockerfile.Args[1].Value, "0")
-							h.AssertEq(t, dockerfile.Args[2].Name, "group_id")
-							h.AssertEq(t, dockerfile.Args[2].Value, "5678")
-							h.AssertEq(t, dockerfile.Args[3], expectedDockerfileA.Args[0])
-
-							return someFakeImage, nil
-						})
-					secondConfig := &v1.ConfigFile{Config: v1.Config{
-						User: "2345:6789",
-						Env:  []string{"SOME_VAR=some-val"},
-					}}
-					someFakeImage.ConfigFileReturnsOnCall(1, secondConfig, nil)
-
-					// second dockerfile
-
-					someFakeImage.ConfigFileReturnsOnCall(2, secondConfig, nil)
-					fakeDockerfileApplier.EXPECT().Apply(
-						gomock.Any(),
-						someFakeImage,
-						extend.Options{
-							BuildContext: "some-app-dir",
-							IgnorePaths:  []string{"some-app-dir", "some-layers-dir", "some-platform-dir"},
-							CacheTTL:     7 * (24 * time.Hour),
-						},
-						logger,
-					).DoAndReturn(
-						func(dockerfile extend.Dockerfile, _ v1.Image, _ extend.Options, _ llog.Logger) (v1.Image, error) {
-							h.AssertEq(t, dockerfile.Path, expectedDockerfileB.Path)
-							h.AssertEq(t, len(dockerfile.Args), 4)
-							h.AssertEq(t, dockerfile.Args[0].Name, "build_id")
-							_, err := uuid.Parse(dockerfile.Args[0].Value)
-							h.AssertNil(t, err)
-							h.AssertEq(t, dockerfile.Args[1].Name, "user_id")
-							h.AssertEq(t, dockerfile.Args[1].Value, "2345")
-							h.AssertEq(t, dockerfile.Args[2].Name, "group_id")
-							h.AssertEq(t, dockerfile.Args[2].Value, "6789")
-							h.AssertEq(t, dockerfile.Args[3], expectedDockerfileB.Args[0])
-
-							return someFakeImage, nil
-						})
-					someFakeImage.ConfigFileReturnsOnCall(3, secondConfig, nil)
+			expectedDockerfile := extend.Dockerfile{
+				Path: dockerfilePath,
+				Args: []extend.Arg{{Name: fmt.Sprintf("arg%s", id), Value: fmt.Sprintf("value%s", id)}},
+			}
+			h.Mkdir(t, basePath)
+			h.Mkfile(t, "some dockerfile content", dockerfilePath)
+
+			buf := new(bytes.Buffer)
+			var data extend.Config
+			switch kind {
+			case "run":
+				data = extend.Config{Run: extend.BuildConfig{Args: expectedDockerfile.Args}}
+			case "build":
+				data = extend.Config{Build: extend.BuildConfig{Args: expectedDockerfile.Args}}
+			default:
+				t.Fail()
+			}
+			switch contextDir {
+			case "specific":
+				h.Mkdir(t, filepath.Join(basePath, fmt.Sprintf("context.%s", kind)))
+			case "shared":
+				h.Mkdir(t, filepath.Join(basePath, "context"))
+			case "app":
+				// nothing to do
+			default:
+				t.Fail()
+			}
```

Comment:
This might read easier if we move it into the Platform API 0.13 block (around line 205)